release/pipline/dags/istio_common_dag.py

```diff
@@ -0,0 +1,251 @@
+
+    MFEST_URL = conf.get('MFEST_URL') or default_conf['MFEST_URL']
+    MFEST_FILE = conf.get('MFEST_FILE') or default_conf['MFEST_FILE']
+    if monthly:
+      MFEST_COMMIT = conf.get('MFEST_COMMIT') or Variable.get(last_daily)
+    else:
+      MFEST_COMMIT = conf.get('MFEST_COMMIT') or default_conf['MFEST_COMMIT']
+    GCR_BUCKET = conf.get('GCR_BUCKET') or default_conf['GCR_BUCKET']
+    GCS_BUCKET = conf.get('GCS_BUCKET') or default_conf['GCS_BUCKET']
+    GCS_PATH = conf.get('GCS_PATH') or default_conf['GCS_PATH']
+    GCS_DEST = conf.get('GCR_DEST') or default_conf['GCR_DEST']
+
+    SVC_ACCT = conf.get('SVC_ACCT') or default_conf['SVC_ACCT']
+    GITHUB_ORG = conf.get('GITHUB_ORG') or default_conf['GITHUB_ORG']
+    GITHUB_REPO = conf.get('GITHUB_REPO') or default_conf['GITHUB_REPO']
+    GCS_GITHUB_PATH = conf.get(
+        'GCS_GITHUB_PATH') or default_conf['GCS_GITHUB_PATH']
+
+    TOKEN_FILE = conf.get('TOKEN_FILE') or default_conf['TOKEN_FILE']
+    GCR_DEST = conf.get('GCR_DEST') or default_conf['GCR_DEST']
+    GCS_DEST = conf.get('GCS_DEST') or default_conf['GCS_DEST']
+    DOCKER_HUB = conf.get('DOCKER_HUB') or default_conf['DOCKER_HUB']
+    BUILD_GCS_BUCKET = conf.get(
+        'BUILD_GCS_BUCKET') or default_conf['BUILD_GCS_BUCKET']
+    return {
+        'PROJECT_ID': PROJECT_ID,
+        'MFEST_URL': MFEST_URL,
+        'MFEST_FILE': MFEST_FILE,
+        'MFEST_COMMIT': MFEST_COMMIT,
+        'execution_date': date,
+        'VERSION': VERSION,
+        'VERSION_TUPLE': (major_version, minor_version, patch, date),
+        'GCR_BUCKET': GCR_BUCKET,
+        'GCS_BUCKET': GCS_BUCKET,
+        'GCS_PATH': GCS_PATH,
+        'BUILD_GCS_BUCKET': BUILD_GCS_BUCKET,
+        'BUILD_GCS_PATH': '{}/{}'.format(BUILD_GCS_BUCKET, GCS_PATH),
+        'GCS_SOURCE': '{}/{}'.format(BUILD_GCS_BUCKET, GCS_PATH),
+        'GCR_DEST': GCR_DEST,
+        'GCS_DEST': GCS_DEST,
+        'SVC_ACCT': SVC_ACCT,
+        'GITHUB_ORG': GITHUB_ORG,
+        'GITHUB_REPO': GITHUB_REPO,
+        'GCS_GITHUB_PATH': GCS_GITHUB_PATH,
+        'TOKEN_FILE': TOKEN_FILE,
+        'Docker_HUB': DOCKER_HUB,
+    }
+
+
+
+  def GetSettingTemplate(setting):
+    return "{{ task_instance.xcom_pull(task_ids='generate_workflow_args').%s }}" % (
+        setting)
+
+  generate_flow_args = PythonOperator(
+      task_id='generate_workflow_args',
+      python_callable=GenerateTestArgs,
+      provide_context=True,
+      dag=common_dag,
+  )
+
+  def GetSettingPython(ti, setting):
+    return it.xcom_pull(task_id=generate_flow_args.task_id)[setting]
+
+  get_git_commit_cmd = """
+    {% set settings = task_instance.xcom_pull(task_ids='generate_workflow_args') %}
+    git config --global user.name "TestRunnerBot"
+    git config --global user.email "testrunner@example.com"
+    git clone {{ settings.MFEST_URL }} green-builds || exit 2
+    cd green-builds
+    git checkout {{ settings.MFEST_COMMIT }} || exit 3
+    git rev-parse HEAD
```

Comment:
A potential alternative is to query something like:
https://api.github.com/repos/istio/green-builds/git/refs/heads/master
though a different URL would be needed for branch vs commit, and I'm not sure what to do about the branch@date that was being attempted (or if it should stay).

Reply:
I prefer using pure git to github apis. This method does not count aginst our github api count. It could be used to copy the time appropriate scripts (I don't think we want to see below). The branch@date is key to pipeline backfilling and repeatability. This method allows settings.MFEST_COMMIT to be anything supported by git checkout which is good and logical behavior